Userland/Libraries/LibCore/Account.cpp

```diff
@@ -294,11 +294,17 @@ ErrorOr<ByteString> Account::generate_shadow_file() const
 {
     StringBuilder builder;
 
-    setspent();
+    FILE* shadow_file = fopen("/etc/shadow", "r");
+    if (!shadow_file)
+        return Error::from_errno(errno);
+
+    ScopeGuard file_guard { [&] {
+        fclose(shadow_file);
+    } };
```

Comment:
Shouldn't we have a `Core::File` which auto closes?

Reply:
Ah yeah, no clue

Always forget that FILE* and a FD are two different things for some reason